src/basics.php

```diff
@@ -62,6 +62,31 @@ function debug($var, $showHtml = null, $showFrom = true)
 
 }
 
+if (!function_exists('warn')) {
+    /**
+     * Throws a warning.
+     *
+     * @param string $message The warning message.
+     * @return void
+     */
+    function warn($message)
```

Comment:
This function would need to be in `src/Core/functions.php` as the cache package is subsplit out, and cakephp/core does not include this file.